### DISCUSSION TITLE
Fixed windows

### DISCUSSION THREAD

**Discussion by s-kuberski** [fjosw/pyerrors#233] [Ideas]
When having one observable with large autocorrelation which has been computed on every nth configuration and another observable that is less correlated and has been computed on all configurations, the autocorrelation function of a derived observable of the two can have a surprising pattern, in the sense that it is only significantly different from zero on separations that are a multiple of n. The automatic windowing procedure fails in this case. This is not an issue per se and expected.

Currently, it is not really possible to deal with this case in pyerrors, because the only parameters for the gamma method are 'S' and 'tau_exp'. It could be helpful to allow to set either a fixed or a minimal window size, either when calling the gamma method, or via a global parameter, just as it is the case for 'S' and 'tau_exp'.

This is an edge case, but the extension of the package in this regard would not interfere with more standard cases. Before implementing this, I wanted to ask you for your feedback on this proposal. I would prefer a minimal window as this option guarantees that the only thing that can happen in standard cases is that the statistical uncertainty of the estimateof tau_int  is enlarged, whereas the systematic one might be larger in the case of a fixed length (per ensemble for all observables).

**Comment by fjosw** [comment]
I like the idea of having another option to determine the window and I think we considered this already at some point in the past but never implemented it. Only specifying a minimal window sounds like a good plan. I guess the risk with that option is that the automatic windowing procedure can become unstable when the minimal window is chosen too large which could significantly overestimate the error in such situations.

**Reply by s-kuberski** [reply]
Thanks for the feedback. I agree that error estimates might be affected if the minimal window is chosen too large. I'd expect that the automatic windowing procedure would stop immediately at the minimal window, but of course there can be fluctuations in the autocorrelation function and its error estimate such that issues could appear. (Then, if you choose `S` stupidly, something can break, as well)
I'll implement a first version and I'll play a bit around with it.
